.github/workflows/submitty_ci_2.yml

```diff
@@ -0,0 +1,216 @@
+name: Submitty CI-2
```

Comment:
Many of the steps here do multiple different things. We should break this up into individual run commands, leaving groupings together as makes sense. For example, we should run all pip installs in one run, all apt-gets in one run, etc., while then doing stuff like "setup and populate DB" would be its own step, running the setup scripts can be its own step, etc. I can give more concrete examples per run command if so desired.

Doing this will make it easier at a glance to tell what went wrong and debug. As GitHub does not have a sense of "pipeline steps" like Travis does, we really do just want to get fine-grained.

Reply:
When you say remove individual run commands do you mean get rid of the `run |` and replace it with a single line 
`run` ?

Reply:
Changing:

```
- name: Install Submitty and populate database
            run: |
              PGPASSWORD=${PGPASSWORD} psql -d postgres -h localhost -U postgres -c "CREATE ROLE submitty_dbuser WITH SUPERUSER CREATEDB CREATEROLE LOGIN PASSWORD 'submitty_dbuser'"
              PGPASSWORD=${PGPASSWORD} psql -d postgres -h localhost -U submitty_dbuser -c "CREATE DATABASE submitty"
              sudo apt-get update
              sudo apt-get install libseccomp-dev
              sudo apt-get install libboost-all-dev
              sudo apt-get install apache2
              sudo apt-get install apache2-suexec-custom
              sudo apt-get install libapache2-mod-authnz-external
              sudo apt-get install libapache2-mod-authz-unixgroup
              sudo apt-get install libapache2-mod-wsgi-py3
              sudo apt-get install nginx
              sudo apt-get install php${PHP_VER}-fpm
              export SUBMITTY_INSTALL_DIR=/usr/local/submitty #need to export otherwise shell scripts don't have access
              export SUBMITTY_REPOSITORY=${SUBMITTY_INSTALL_DIR}/GIT_CHECKOUT/Submitty
              pushd ${SUBMITTY_INSTALL_DIR}/GIT_CHECKOUT/Submitty
              sudo -E env "PATH=$PATH" bash .setup/testing/autograder.sh
              sudo -E env "PATH=$PATH" bash .setup/testing/setup.sh
              popd
```

to be more like:

```
- name: Setup Database
  run: |
      PGPASSWORD=${PGPASSWORD} psql -d postgres -h localhost -U postgres -c "CREATE ROLE submitty_dbuser WITH SUPERUSER CREATEDB CREATEROLE LOGIN PASSWORD 'submitty_dbuser'"
      PGPASSWORD=${PGPASSWORD} psql -d postgres -h localhost -U submitty_dbuser -c "CREATE DATABASE submitty"

- name: Install apt-get packages
  run: |
      sudo apt-get update
      sudo apt-get install libseccomp-dev
      sudo apt-get install libboost-all-dev
      sudo apt-get install apache2
      sudo apt-get install apache2-suexec-custom
      sudo apt-get install libapache2-mod-authnz-external
      sudo apt-get install libapache2-mod-authz-unixgroup
      sudo apt-get install libapache2-mod-wsgi-py3
      sudo apt-get install nginx
      sudo apt-get install php${PHP_VER}-fpm

- name: Setup third party autograding dependencies
  run: sudo -E env "PATH=$PATH" bash .setup/testing/autograder.sh

- name: Populate Submitty
  run: sudo -E env "PATH=$PATH" bash .setup/testing/setup.sh
```

Each of these commands are individual from each other and do not rely on exports or anything from the previous step. Separating them makes it easier to say why a build failed (was it that the database failed to initialize? apt-get install error? etc) without having to dig into the full log for that line.

Reply:
Ok makes sense, thanks